pkg/services/libraryelements/api.go

```diff
@@ -244,7 +330,50 @@ func (l *LibraryElementService) getByNameHandler(c *contextmodel.ReqContext) res
 		return toLibraryElementError(err, "Failed to get library element")
 	}
 
-	return response.JSON(http.StatusOK, model.LibraryElementArrayResponse{Result: elements})
+	filteredElements := make([]model.LibraryElementDTO, 0)
+	for _, el := range elements {
+		allowed, err := l.evalPermission(c, "read", el.UID, el.FolderID)
+		if err != nil {
+			return toLibraryElementError(err, err.Error())
+		}
+		if allowed {
+			filteredElements = append(filteredElements, el)
+		}
+	}
+
+	return response.JSON(http.StatusOK, model.LibraryElementArrayResponse{Result: filteredElements})
+}
+
+func (l *LibraryElementService) evalPermission(c *contextmodel.ReqContext, action string, elementUID string, folderID int64) (bool, error) {
+	return l.AccessControl.Evaluate(
+		c.Req.Context(),
+		c.SignedInUser,
+		ac.EvalAny(
+			ac.EvalPermission(
+				ScopeLibraryPanelsRoot+":"+action,
```

Comment:
Nit: you could also use a scope provider pattern here: https://github.com/grafana/grafana/blob/main/pkg/services/accesscontrol/scope.go#L95

Reply:
Hmm, I wanted to do this, but the library element service's `getLibraryElementByUid` method expects a pointer to a `user.SignedInUser`. Do you know if there's a way to access the currently signed-in user struct from a `ScopeAttributeResolverFunc`?